### ISSUE TITLE
Graduated Filter interferes with "L*a*b* Adjustment" contrast setting

### ISSUE THREAD

**Issue by da-phil** [Beep6581/RawTherapee#2976]
When you use choose to increase the contrast in the "L*a*b* Adjustment" setting and activate the graduated filter later you'll notice that the supposedly unaffected area of the graduated filter becomes brighter once you turned on the grad filter. However lightness and cromaticity settings in the same dialoge are not affected by activating the grad filter.

I'm using the latest git version from master branch:
```
Branch: master
Version: 4.2.476
Changeset: 13febd8009c5b1504e08946b76d7a8f15f36ebfc
Compiler: cc 4.8.4
Processor: x86_64
System: Linux
Bit depth: 64 bits
Gtkmm: V2.24.4
Build type: release
Build flags:   -Werror=unused-label -fopenmp -Werror=unknown-pragmas -O3 -DNDEBUG
Link flags:    
OpenMP support: ON
MMAP support: ON
```

I guess it's faily easy to reproduce, just take any image and follow the steps above. If you're having trouble to reproduce the problem I can provide screenshots and *.pp3s.

Cheers
Phil

**Comment by Beep6581** [comment]
Confirmed. Looks like L*a*b* Adjustments are after the Graduated Filter in the pipeline, which I think is wrong. This is one of several tools that need to be moved to a different place in the pipeline.

**Comment by da-phil** [comment]
Yep, but strangely enough the lightness and cromaticity settings in the very same dialoge are not affected, they should be also in the same place like contrast setting in the pipeline, aren't they?

**Comment by michaelezra** [comment]
It is a beneficial feature of the graduated filter to act as a graduated exposure adjustment allowing to recover highlights, not just darkening/lightening. Since exposure changes are early in pipeline, may be thats where the graduated filter should also be?

**Comment by da-phil** [comment]
I think that the graduated filter is only related to exposure changes (that at least was the original purpose of hardware grads) and that's why it might make sense to apply this filter immediately after the base exposure changes in the pipeline, doesn't it?
